### DISCUSSION TITLE
Unable to start language server (probably configuration rather than bug)

### DISCUSSION THREAD

**Discussion by njlivesey** [interactive-data-language/vscode-idl#12] [Q&A] [answered]
I’m having issues with IDL for VSCode.  Things generally appear to be working to a fair degree, but there are error messages on startup which indicate that things are not happy.  It says that the IDL Language Server crashed.  So far as I can tell I do have node.js installed and in my path.  I’m not familiar with Javascript, nor am I that familiar with VSCode, so apologies if the issue is something basic.

I’ve pasted the logs below, as well as a screenshot of the specific message.  In case it’s relevant, I’m running VSCode (1.83.1, Universal) on my Mac (MacOS 13.6), connecting via ssh to a linux server (I’ve pasted /etc/os-release below too).

Any pointers are welcome.

Yours,

Nathaniel Livesey
NASA JPL.

idl-client info IDL directory and path: "/opt/local/depot/harris/idl/bin"
  []
idl-client info Registering client lib commands
idl-client info Registering code commands
idl-client info Trying to detect node.js
idl-client info Starting the language server using "node". Output from "node --version":
  v6.13.1
idl-client info Starting language server
/users/livesey/.vscode-server/extensions/idl.idl-for-vscode-3.2.3/dist/apps/server/main.js:1
(function (exports, require, module, __filename, __dirname) { (()=>{var e=[,(e,t,n)=>{"use strict";Object.defineProperty(t,"__esModule",{value:!0});const o=n(2);o.__exportStar(n(3),t),o.__exportStar(n(680),t)},(e,t,n)=>{"use strict";n.r(t),n.d(t,{__addDisposableResource:()=>M,__assign:()=>i,__asyncDelegator:()=>v,__asyncGenerator:()=>y,__asyncValues:()=>R,__await:()=>L,__awaiter:()=>E,__classPrivateFieldGet:()=>D,__classPrivateFieldIn:()=>x,__classPrivateFieldSet:()=>C,__createBinding:()=>h,__decorate:()=>a,__disposeResources:()=>w,__esDecorate:()=>l,__exportStar:()=>m,__extends:()=>r,__generator:()=>f,__importDefault:()=>P,__importStar:()=>I,__makeTemplateObject:()=>b,__metadata:()=>_,__param:()=>c,__propKey:()=>p,__read:()=>T,__rest:()=>s,__runInitializers:()=>u,__setFunctionName:()=>d,__spread:()=>N,__spreadArray:()=>S,__spreadArrays:()=>g,__values:()=>O,default:()=>U});var o=function(e,t){return o=Object.setProt

SyntaxError: Unexpected token function
    at createScript (vm.js:56:10)
    at Object.runInThisContext (vm.js:97:10)
    at Module._compile (module.js:549:28)
    at Object.Module._extensions..js (module.js:586:10)
    at Module.load (module.js:494:32)
    at tryModuleLoad (module.js:453:12)
    at Function.Module._load (module.js:445:3)
    at Module.runMain (module.js:611:10)
    at run (bootstrap_node.js:387:7)
    at startup (bootstrap_node.js:153:9)
[Error - 9:07:43 AM] Server process exited with code 1.
idl-client error The connection to the IDL Language Server unexpectedly closed
idl-client error Error initializing IDL Language Server
  {
    message: [ 'Pending response rejected since connection got disposed' ],
    stack: [
      'Error: Pending response rejected since connection got disposed',
      '\tat Object.dispose (/users/livesey/.vscode-server/extensions/idl.idl-for-vscode-3.2.3/dist/apps/client/main.js:1:388302)',
      '\tat Object.dispose (/users/livesey/.vscode-server/extensions/idl.idl-for-vscode-3.2.3/dist/apps/client/main.js:1:701204)',
      '\tat LanguageClient.handleConnectionClosed (/users/livesey/.vscode-server/extensions/idl.idl-for-vscode-3.2.3/dist/apps/client/main.js:1:701610)',
      '\tat LanguageClient.handleConnectionClosed (/users/livesey/.vscode-server/extensions/idl.idl-for-vscode-3.2.3/dist/apps/client/main.js:1:668858)',
      '\tat /users/livesey/.vscode-server/extensions/idl.idl-for-vscode-3.2.3/dist/apps/client/main.js:1:701353',
      '\tat CallbackList.invoke (/users/livesey/.vscode-server/extensions/idl.idl-for-vscode-3.2.3/dist/apps/client/main.js:1:359060)',
      '\tat Emitter.fire (/users/livesey/.vscode-server/extensions/idl.idl-for-vscode-3.2.3/dist/apps/client/main.js:1:359848)',
      '\tat te (/users/livesey/.vscode-server/extensions/idl.idl-for-vscode-3.2.3/dist/apps/client/main.js:1:376566)',
      '\tat CallbackList.invoke (/users/livesey/.vscode-server/extensions/idl.idl-for-vscode-3.2.3/dist/apps/client/main.js:1:359060)',
      '\tat Emitter.fire (/users/livesey/.vscode-server/extensions/idl.idl-for-vscode-3.2.3/dist/apps/client/main.js:1:359848)',
      '\tat StreamMessageWriter.fireClose (/users/livesey/.vscode-server/extensions/idl.idl-for-vscode-3.2.3/dist/apps/client/main.js:1:368517)',
      '\tat Socket.<anonymous> (/users/livesey/.vscode-server/extensions/idl.idl-for-vscode-3.2.3/dist/apps/client/main.js:1:369316)',
      '\tat Socket.emit (node:events:513:28)',
      '\tat Socket.emit (node:domain:489:12)',
      '\tat Pipe.<anonymous> (node:net:322:12)'
    ]
  }
[Error - 9:07:43 AM] Connection to server got closed. Server will not be restarted.
[Error - 9:07:43 AM] idl client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
idl-client info Registering debug configuration provider
idl-client info Starting IDL debug server (not IDL)
idl-client info Registering debug commands
idl-client info Creating IDL tree view
idl-client info Registering webview commands
idl-client info Registering notebook serializer and controller
idl-client info Creating notebook renderer messenger
idl-client info Registering notebook commands
idl-client info Registering ENVI opener
idl-client info Registering docs commands

==================================================

(njl-base-3.10-18oct23) 2023d293 mlsscf-share1 % cat /etc/os-release 
NAME="Oracle Linux Server"
VERSION="8.8"
ID="ol"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="8.8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Oracle Linux Server 8.8"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:oracle:linux:8:8:server"
HOME_URL=https://linux.oracle.com/
BUG_REPORT_URL=https://github.com/oracle/oracle-linux

ORACLE_BUGZILLA_PRODUCT="Oracle Linux 8"
ORACLE_BUGZILLA_PRODUCT_VERSION=8.8
ORACLE_SUPPORT_PRODUCT="Oracle Linux"
ORACLE_SUPPORT_PRODUCT_VERSION=8.8

**Comment by znorman-harris** [comment]
Hey There!

Reading through the logs, it looks like it is trying to use an older version of node.js (you have version 6 on your system path).

Let me see if I can add in a filter to check the version and ignore node.js if the version is too low which will cause problems.

As a workaround, if you can remove node.js from your path, then that should hopefully work around the problem until we can get a check/fix in.

**Comment by njlivesey** [comment]
Hi Zac,

   Many thanks for following up.  Ah, that’s interesting.  It seems to be the version supplied by my Anaconda environment.  Perhaps I’ll try updating that (though that can give its own problems as perhaps something else requires a version that old).

Nathaniel

Nathaniel Livesey
Mail Stop 183-701, Jet Propulsion Laboratory
4800 Oak Grove Drive, Pasadena, California 91109.
Phone: +1 818 354-4214<tel:+18183544214>   Fax: +1 818 393-5065<tel:+18183935065>

From: Zachary Norman ***@***.***>
Date: Tuesday, October 24, 2023 at 8:48 AM
To: interactive-data-language/vscode-idl ***@***.***>
Cc: Livesey, Nathaniel J (US 3290) ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [interactive-data-language/vscode-idl] Unable to start language server (probably configuration rather than bug) (Discussion #12)

Hey There!

Reading through the logs, it looks like it is trying to use an older version of node.js (you have version 6 on your system path).

Let me see if I can add in a filter to check the version and ignore node.js if the version is too low which will cause problems.

As a workaround, if you can remove node.js from your path, then that should hopefully work around the problem until we can get a check/fix in.

**Comment by znorman-harris** [comment, marked as answer]
Alright, I just published a new version of the extension that should ignore any node.js versions less than v16.

If this doesn't fix the problem, let me know and I can poke at it some more!

**Comment by znorman-harris** [comment]
Just checking in, did the extension update fix the problem for you?

**Comment by njlivesey** [comment]
Hi Zach,

   Thanks so much for following up and sorry for not getting back to you myself.  Yes, I believe it did.

Many thanks, again,

Nathaniel

Nathaniel Livesey
Mail Stop 183-701, Jet Propulsion Laboratory
4800 Oak Grove Drive, Pasadena, California 91109.
Phone: +1 818 354-4214<tel:+18183544214>   Fax: +1 818 393-5065<tel:+18183935065>

From: Zachary Norman ***@***.***>
Date: Monday, October 30, 2023 at 8:23 AM
To: interactive-data-language/vscode-idl ***@***.***>
Cc: Livesey, Nathaniel J (US 3290) ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [interactive-data-language/vscode-idl] Unable to start language server (probably configuration rather than bug) (Discussion #12)

Just checking in, did the extension update fix the problem for you?

**Comment by znorman-harris** [comment]
Great to hear! just wanted to check. I'll mark my response above as the answer and close.
